src/webgpu/api/operation/command_buffer/copyTextureToTexture.spec.ts

```diff
@@ -75,9 +223,92 @@ g.test('t2t_non_compressed_color_formats')
             srcTextureSize: { width: 32, height: 32, depth: 1 },
             dstTextureSize: { width: 63, height: 61, depth: 1 },
           },
+        ])
+      )
+      .combine(poptions('format', kRegularTextureFormats))
+      .combine(
+        poptions('copyBoxOffsets', [
+          // From (0, 0) of src to (0, 0) of dst.
           {
-            srcTextureSize: { width: 32, height: 32, depth: 3 },
-            dstTextureSize: { width: 32, height: 32, depth: 3 },
+            srcOffset: { x: 0, y: 0, z: 0 },
+            dstOffset: { x: 0, y: 0, z: 0 },
+            copyExtent: { width: 0, height: 0, depth: 0 },
+          },
+          // From (0, 0) of src to (1, 0) of dst.
+          {
+            srcOffset: { x: 0, y: 0, z: 0 },
+            dstOffset: { x: 1, y: 0, z: 0 },
+            copyExtent: { width: 0, height: 0, depth: 0 },
+          },
+          // From (0, 0) of src to (0, 1) of dst.
+          {
+            srcOffset: { x: 0, y: 0, z: 0 },
+            dstOffset: { x: 0, y: 1, z: 0 },
+            copyExtent: { width: 0, height: 0, depth: 0 },
+          },
+          // From (1, 0) of src to (0, 0) of dst.
+          {
+            srcOffset: { x: 1, y: 0, z: 0 },
+            dstOffset: { x: 0, y: 0, z: 0 },
+            copyExtent: { width: 0, height: 0, depth: 0 },
+          },
+          // From (0, 1) of src to (0, 0) of dst.
+          {
+            srcOffset: { x: 0, y: 1, z: 0 },
+            dstOffset: { x: 0, y: 0, z: 0 },
+            copyExtent: { width: 0, height: 0, depth: 0 },
+          },
+          // From (1, 0) of src to (0, 0) of dst, and the copy extent will not cover the last
+          // column of both source and destination texture.
+          {
+            srcOffset: { x: 1, y: 0, z: 0 },
+            dstOffset: { x: 0, y: 0, z: 0 },
+            copyExtent: { width: -1, height: 0, depth: 0 },
+          },
+          // From (0, 1) of src to (0, 0) of dst, and the copy extent will not cover the last
+          // row of both source and destination texture.
+          {
+            srcOffset: { x: 0, y: 1, z: 0 },
+            dstOffset: { x: 0, y: 0, z: 0 },
+            copyExtent: { width: 0, height: -1, depth: 0 },
+          },
```

Comment:
Add no-op copy cases to this list (that are not factored out into the common cases)?

Reply:
I've put all the zero-copy cases into one test group "zero_copies", PTAL.